From 948f9bebfc880dd7586cfa44fcd53e61e95288b6 Mon Sep 17 00:00:00 2001
From: Tristan Seifert <tristan@tseifert.me>
Date: Wed, 22 Jun 2022 19:41:15 -0700
Subject: [PATCH 3/3] Add eastrising lcd

---
 drivers/gpu/drm/panel/panel-simple.c | 35 ++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 8dd7013c75f2..1aa786ace35c 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -4466,7 +4466,42 @@ static const struct panel_desc arm_rtsm = {
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 };
 
+
+// VFP >= 2, VBP >= 5
+// HFP >= 2, HBP >= 5
+static const struct drm_display_mode eastrising_er_tft040_1_mode = {
+	.clock = 25000,
+	.hdisplay = 480,
+	.hsync_start = 480 + 10,
+	.hsync_end = 480 + 10 + 10,
+	.htotal = 480 + 10 + 10 + 15,
+	.vdisplay = 800,
+	.vsync_start = 800 + 28,
+	.vsync_end = 800 + 28 + 3,
+	.vtotal = 800 + 28 + 3 + 28,
+};
+
+static const struct panel_desc eastrising_er_tft040_1 = {
+	.modes = &eastrising_er_tft040_1_mode,
+	.num_modes = 1,
+	.bpc = 8,
+	.size = {
+		.width = 86,
+		.height = 52,
+	},
+
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+	.bus_flags = DRM_BUS_FLAG_DE_HIGH | DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE |
+		     DRM_BUS_FLAG_SYNC_DRIVE_NEGEDGE,
+	.connector_type = DRM_MODE_CONNECTOR_DPI,
+};
+
+
 static const struct of_device_id platform_of_match[] = {
+    {
+        .compatible = "eastrising,er-tft040-1",
+        .data = &eastrising_er_tft040_1,
+    },
 	{
 		.compatible = "ampire,am-1280800n3tzqw-t00h",
 		.data = &ampire_am_1280800n3tzqw_t00h,
-- 
2.32.1 (Apple Git-133)


